src/metabase/driver/sql_jdbc/actions.clj

```diff
@@ -14,29 +14,32 @@
             [metabase.query-processor.store :as qp.store]
             [metabase.util :as u]
             [metabase.util.honeysql-extensions :as hx]
-            [metabase.util.i18n :refer [tru]]
+            [metabase.util.i18n :refer [tru trs]]
             [toucan.db :as db])
   (:import java.sql.Connection))
 
 (defmulti parse-sql-error
   "Parses the raw error message returned after an error in the driver database occurs, and converts it into a sequence
   of maps with a :column and :message key indicating what went wrong."
-  {:arglists '([driver ^java.sql.Connection connection e]), :added "0.44.0"}
+  {:arglists '([driver database e]), :added "0.44.0"}
```

Comment:
I tweaked `parse-sql-error` to take the original Database rather than a `java.sql.Connection`. The expectation is that someone will use `sql-jdbc.conn/db->pooled-connection-spec` to get a connection from the pool rather than using the current `Connection`. Why? because the current Connection might be inside a transaction and in an error state, which means future queries against it will fail with some sort of "transaction already aborted" error. So it's better to query this information from a separate Database connection.